components/button/test/button-toggle.test.js

```diff
@@ -63,4 +63,47 @@ describe('d2l-button-toggle', () => {
 
 	});
 
+	describe('consumer manages state', () => {
+
+		it('click with no state management', async() => {
+			const el = await fixture(html`
+				<d2l-button-toggle>
+					<d2l-button-icon slot="not-pressed" icon="tier1:pin-hollow" text="Unpinned, click to pin."></d2l-button-icon>
+					<d2l-button-icon slot="pressed" icon="tier1:pin-filled" text="Pinned, click to unpin."></d2l-button-icon>
+				</d2l-button-toggle>
+			`);
+			const buttonIcons = el.querySelectorAll('d2l-button-icon');
+			buttonIcons[0].addEventListener('click', (e) => {
+				e.preventDefault();
+			});
+			buttonIcons[1].addEventListener('click', (e) => {
+				e.preventDefault();
+			});
+			clickElem(el.querySelector('[slot="not-pressed"]'));
+			await aTimeout(100);
```

Comment:
Is the timeout needed? Like if you remove the timeout and comment out the code that checks whether default was prevented, does the test fail like it should? Since `clickElem`/etc. are `async` I've typically seen that by the time they finish, any changes have been applied. But yeah, it can be tricky to ensure that a thing _hasn't_ happened and isn't going to haha.

Reply:
It does seem somewhat needed. Without it I get `Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`. With the timeout smaller and the check code commented out it does fail as expected.
Yeah it's that weird case of wanting to make sure it hasn't happened. I can reduce the timeout though; that seems probably safe.

Reply:
Ah right, what about if you `await clickElem(...)`?

Reply:
Ah yup that passes/fails as expected. I can update to use that.